### PR TITLE
[ONNX] Adding 'numel' and 'to' export for script module

### PR DESCRIPTION
**Pull request by neginraoof**
These two ops are needed for torchvision model export. Since we're scripting a part of the code for dynamic export of models (in https://github.com/pytorch/vision/pull/2052), these two changes are requited.

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1505,6 +1505,15 @@ def sort(g, self, dim, decending, out=None):
 
     return g.op("TopK", self, k_i=self.type().sizes()[dim], axis_i=dim, outputs=2)
 
+
+def numel(g, self):
+    shape = g.op("Shape", self)
+    dtype = shape.type().scalarType()
+    if dtype is None:  # dtype could be None for a script module
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:1512]: `shape` is just created above, I think here `dtype` would always `None`. 

On the other hand, according to onnx spec, `Shape` operator always output tensor of type `Long`, why do we need to insert the `cast` node?

Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py:1512]: I was seeing the ONNXRuntime error for kernel not implemented, and I thought this missing type is the issue. But that's correct, it should not be neede.
This PR needs to wait until https://github.com/microsoft/onnxruntime/pull/3507 is merged then.

**Review by lara-hdr — APPROVED** [review]

**Comment by lara-hdr** [comment]
@neginraoof, it seems like ReduceProd is not available in ort for all opsets (it's erroring out with "Failed to find kernel for ReduceProd" for opsets 7-9). Could you disable the tests for the older versions and/or request the kernel implementation?

**Comment by neginraoof** [comment]
@lara-hdr Thanks for reviewing this. I update onnxruntime ReduceProd, waiting for nightly build (it's stuck), I need to update ort-nightly version after that.

**Comment by neginraoof** [comment]
cc @houseroad for review. Thanks.

**Comment by neginraoof** [comment]
cc @houseroad for review. Thanks.

**Review by BowenBao — APPROVED** [review]
LGTM

**Comment by neginraoof** [comment]
Thanks @BowenBao 
@houseroad if there are no internal failures, can we merge this please?
Thanks.

**Review by houseroad — APPROVED** [review]

**Pull request closed without merging**